spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClient.java

```diff
@@ -93,12 +110,18 @@ public String description() {
 		return instances;
 	}
 
-	private Map<String, String> getLabels(String serviceName) {
-		final Service service = client.services().withName(serviceName).get();
-		if (service != null) {
-			return service.getMetadata().getLabels();
+	// always returns a new map that contain all the entries of the original map
+	// but with the keys prefixed
+	private Map<String, String> getMapWithPrefixedKeys(Map<String, String> map, String prefix) {
+		// when the prefix is empty just return an map with the same entries
+		if (prefix.isEmpty()) {
+			return new HashMap<>(map);
```

Comment:
Why a new map?

Reply:
I return a new map to keep the method behavior consistent in all cases - it always returns a new map regardless of the input. That way if it's ever used in another invocation, it's crystal clear how it behaves and doesn't present any surprises.

That said, I have no problem changing it if you like :)

Reply:
It's return value is input to `metadata.putAll()`. May as well not do extra work.

Reply:
Fixed